### PR TITLE
Add python-dotenv to dev dependencies

### LINKED ISSUE #7029: [main, [dev] install] WARNING: pydantic 2.7.4 does not provide the extra 'dotenv'
### 🐛 Bug Report

Made a fresh python 3.12 conda env.
Fresh main, pulled from upstream.
pip install -e .[dev]
Getting `WARNING: pydantic 2.7.4 does not provide the extra 'dotenv'` from pip.

### 💡 Steps to Reproduce

Made a fresh python 3.12 conda env.
Fresh main, pulled from upstream.
pip install -e .[dev]

### 💡 Expected Behavior

No warning.

### 🌎 Environment

napari: 0.5.0a2.dev760+g74ecc56c6
Platform: macOS-13.3.1-arm64-arm-64bit
System: MacOS 13.3.1
Python: 3.12.4 | packaged by conda-forge | (main, Jun 17 2024, 10:13:44) [Clang 16.0.6 ]
Qt: 6.7.1
PyQt6: 
NumPy: 2.0.0
SciPy: 1.14.0
Dask: 2024.6.2
VisPy: 0.14.3
magicgui: 0.8.3
superqt: 0.6.7
in-n-out: 0.2.1
app-model: 0.2.7
npe2: 0.7.6

OpenGL:
  - GL version:  2.1 Metal - 83.1
  - MAX_TEXTURE_SIZE: 16384
  - GL_MAX_3D_TEXTURE_SIZE: 2048

Screens:
  - screen 1: resolution 1680x1050, scale 2.0

Optional:
  - numba: 0.60.0
  - triangle not installed

Settings path:
  - /Users/piotrsobolewski/Library/Application Support/napari/napari-dev_a1eb8b76ba95fa16ad06e26097b46b8455dfbf0b/settings.yaml
Plugins:
  - napari: 0.5.0a2.dev760+g74ecc56c6 (81 contributions)
  - napari-console: 0.0.9 (0 contributions)
  - napari-svg: 0.2.0 (2 contributions)

### 💡 Additional Context

_No response_

### PR DESCRIPTION
**Pull request by willingc**
Closes #7029

# Description
Add python-dotenv to dev dependencies.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -161,6 +161,7 @@ dev = [
     "check-manifest>=0.42",
     "pre-commit>=2.9.0",
     "pydantic[dotenv]",
```
Comment (Czaki) [inline pyproject.toml]:
```suggestion
    "pydantic",
```

**Review by Czaki — APPROVED** [review]
As v1 still uses `python-dotenv` https://github.com/pydantic/pydantic/blob/11d04f3602a38e32bca70577e41d05d257f86c45/pydantic/v1/env_settings.py#L330

I think it is valid until we drop pyenv v1 and move to pydantic 2 code

File: pyproject.toml (hunk 1 above)
Comment (Czaki) [reply pyproject.toml]: Without remove it, the warning will persist.

**Review by psobolewskiPhD — APPROVED** [review]

**Comment by psobolewskiPhD** [comment]
Thanks! I never got around to doing it 😬

**Pull request merged**